Brittany Crocker/Week 01/Arrays and Functions/arrays-and-functions.js

```diff
@@ -0,0 +1,151 @@
+// # Array and Functions Bonus Material
+//
+// 1. Define a function `maxOfTwoNumbers` that takes two numbers as arguments and returns the largest of them. Use the if-then-else construct available in Javascript. You'll have to remember your pre-work, or do some googling to figure this out.
+// 2. Define a function `maxOfThree` that takes three numbers as arguments and returns the largest of them.
+// 3. Write a function that takes a character (i.e. a string of length 1) and returns true if it is a vowel, false otherwise.
+// 4. Define a function `sumArray` and a function `multiplyArray` that sums and multiplies (respectively) all the numbers in an array of numbers. For example, `sumArray([1,2,3,4])` should return 10, and `multiplyArray([1,2,3,4])` should return 24.
+
+// maxOfTwoNumbers
+const maxOfTwoNumbers = function(num1, num2) {
+  if (num1 > num2){
+    return num1;
+  } else if (num2 > num1) {
+    return num2;
+  }
+};
+console.log("maxOfTwoNumbers " + maxOfTwoNumbers(12, 20));
+
+// maxOfThreeNumbers
+const maxOfThreeNumbers = function(num1, num2, num3) {
+  if (num1 > num2 && num1 > num3){
+    return num1;
+  } else if (num2 > num1 && num2 > num3) {
+    return num2;
+  } else if (num3 > num1 && num3 > num2){
+    return num3;
+  }
+};
+console.log("maxOfThreeNumbers " + maxOfThreeNumbers(12,200,45));
+
+// isVowel
+const isVowel = function(string) {
+  if (string.includes("a","e","i","o","u")){
+    return true;
+  } else {
+    return false;
+  }
+}
+console.log("isVowel " + isVowel("a"));
+console.log("isVowel " + isVowel("b"));
+
+
+//sumArray
+let total = 0
+const sumArray = function(arr) {
+  for(i = 0; i < arr.length; i++){
+  total += arr[i];
+  console.log(`sumArray = ${total}`)
+};
+};
+sumArray([1,2,3,4]);
+
+
+//multiplyArray
+let sum = 1
+const multiplyArray = function(arr) {
+  for(i = 0; i < arr.length; i++){
+  sum *= arr[i]
+  console.log(`multiplyArray = ${sum}`)
+  }
+};
+multiplyArray([1,2,3,4]);
+
+// ## Bonus
+//
+// 5. Define a function `reverseString` that computes the reversal of a string. For example, reverseString("jag testar") should return the string "ratset gaj".
+const reverseString = function(string){
+  splitString = string.split("");
+  reverseArray = splitString.reverse();
+  joinArray = reverseArray.join("");
+  console.log(joinArray);
+}
+reverseString("hello there");
+
+
+// 6. Write a function `findLongestWord` that takes an array of words and returns the length of the longest one.
+const findLongestWord = function(stringArray){
+  let longestWord =[0];
+  for (i=0; i < stringArray.length; i++) {
+    if (stringArray[i].length > longestWord.length) {
+      longestWord.length = stringArray[i].length;
+      console.log(stringArray[i].length);
+    } else {
+      console.log("doh!");
+    }
+}
+}
+findLongestWord(["golden", "pik", "fountain"]);
```

Comment:
You're on the right track here. However you just want the function to return the length of the longest word. Currently you're returning the length of all of the words (and "doh!"). 
Keep in mind that you can have an if statement without an else at the end of it. 
Give this one another go if you've got some time. Also feel free to check out other people's homework for inspiration if you're stuck (the joy of git is every time you do a git pull upstream master, you'll get a copy of the homework everyone else has uploaded).